src/cli/java/org/commcare/util/screen/EntityScreen.java

```diff
@@ -370,4 +370,8 @@ public boolean referencesContainStep(String stepValue) {
         }
         return false;
     }
+
+    public void updateDatum(CommCareSession session, String input) {
+        session.setEntityDatum(session.getNeededDatum(), input);
```

Comment:
We need to update the `MenuSession` copy of the CommCareSession here which is responsible to advance the navigation rather than the internal EntityScreen which mostly snapshots the session state for which the `EntityScreen` was created for.

Reply:
yes I find it confusing as well, I generally use a mental model of performing any writes to the MenuSession copy and then it doesn't generally matter in practice what you use for reading from the session. Probably we can rename the screen's copy to something like `readOnlySession` to make it more evident ?